From df8770080ad4b262f3ad830b21763e27988bed5e Mon Sep 17 00:00:00 2001
From: Liam Howlett <liam.howlett@oracle.com>
Date: Tue, 10 Jan 2023 18:43:03 +0000
Subject: [PATCH 50/50] userfaultfd: Fix vma iterator conversion.

An extra vma_iter_set() was added to one of the loops over the VMAs.
Remove this set as it is suboptimal and unnecessary.

Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
---
 fs/userfaultfd.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/userfaultfd.c b/fs/userfaultfd.c
index e60f86d6b..55f3af668 100644
--- a/fs/userfaultfd.c
+++ b/fs/userfaultfd.c
@@ -1620,7 +1620,6 @@ static int userfaultfd_unregister(struct userfaultfd_ctx *ctx,
 				break;
 		}
 		if (vma->vm_end > end) {
-			vma_iter_set(&vmi, vma->vm_end);
 			ret = split_vma(&vmi, vma, end, 0);
 			if (ret)
 				break;
-- 
2.39.0.rc2.1.gbd5df96b79

